bool wrong_way
string front_color
